components/service/glean/src/main/java/mozilla/components/service/glean/scheduler/BaselinePingScheduler.kt

```diff
@@ -0,0 +1,135 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.glean.scheduler
+
+import android.view.accessibility.AccessibilityManager
+import android.accessibilityservice.AccessibilityServiceInfo
+import android.content.Context
+import android.content.SharedPreferences
+import android.os.Build
+import mozilla.components.service.glean.metrics.GleanBaseline
+import mozilla.components.support.base.log.logger.Logger
+import java.util.concurrent.TimeUnit
+
+/**
+ * BaselinePingScheduler facilitates sending and scheduling of baseline pings when the application
+ * enters the background state (see [GleanLifecycleObserver]).  Using a [PING_INTERVAL_HOURS] of
+ * zero means that there is no restriction on how many pings that can be sent per time period.
+ */
+internal class BaselinePingScheduler(val applicationContext: Context) {
+    private val logger = Logger("glean/BaselinePingScheduler")
+    internal val sharedPreferences: SharedPreferences by lazy { getSharedPrefs() }
```

Comment:
I think we shouldn't pay the I/O price for something we don't use right now. Let's get rid of these prefs